### DISCUSSION TITLE
Suspicious Data Sources for Librespot - Solved

### DISCUSSION THREAD

**Discussion by ndormann** [librespot-org/librespot#971] [General]
I was checking my raspberry pi's firewall for suspicious connections and found a connection to 124.65.199.104 which appears to be in Beijing (BJ) - China (CN). I blocked any traffic to this side with ufw and restarted the service. This time netstat shows a connection attempt that fails. But music playback is working flawlessly. 
So my question: What kind of data sources does librespot (with spotify) use and why. And especially why would it use servers in China when running in Europe? Wouldn't any loadbalancing rather prefer server that are closer?
I hope somebody can help me. Thanks in advance.

**Comment by roderickvd** [comment]
librespot queries the Spotify infrastructure which access point to use. If you check the (verbose) logs you will see which access point it picked. You can try resolving this hostname yourself and see which IP address it points to.

Presumably Spotify's access point resolver will use a combination of geographic + round robin load balancing and filter any servers which it knows are down.

You can even just access the resolver from the browser: http://apresolve.spotify.com is what Spotify uses.

**Reply by ndormann** [reply]
Thanks, it is still weird, that it would use a data source from china. I think there should be enough servers in Europe instead.

**Reply by ndormann** [reply]
Okay I am stupid... This is the hostname with the reverse IP Lookup... So its a DNS and not a IP. I should have recognized that .bc. sooner... its acutally a IP Address from the States. Thanks for your help.

**Comment by kingosticks** [comment]
How did you ascertain that connection came from librespot?

**Reply by ndormann** [reply]
`netstat -e | grep 4070` gives 
```
Proto Recv-Q Send-Q Local Address           Foreign Address         State       User       Inode
tcp        0      0 10.149.21.140:40914     124.65.199.104.bc.:4070 ESTABLISHED 62638      39943 
```
and `ps -u 62638` 
```
    PID TTY          TIME CMD
   2115 ?        00:00:03 librespot
```
shows that this is the librespot user id.
Its the correct port as well.
I don't know what this .bc. means though?

**Reply by kingosticks** [reply]
So Yeh, as you've said in the other thread, this is a hostname. `sudo netstat -anp | grep librespot` next time.

**Reply by ndormann** [reply]
Thanks
